sqlmesh/core/snapshot/evaluator.py

```diff
@@ -262,7 +262,10 @@ def create(
         for snapshot in target_snapshots:
             if not snapshot.is_model or snapshot.is_symbolic:
                 continue
-            for is_deployable in (True, False):
+            is_snapshot_deployable = (
+                deployability_index.is_deployable(snapshot) if deployability_index else True
+            )
+            for is_deployable in self._table_deployability_flags(snapshot, is_snapshot_deployable):
```

Comment:
I think we always want to check for the deployable table here and only sometimes for non-deployable too. Otherwise, this update strips sqlmesh of its self-healing properties, i.e. automatically recreating the table if one was deleted.

Reply:
Right I hadn't considered this possibility. I've updated to always check for deployable versions and when necessary for non-deployable versions. But I now realise this information of missing tables also needs to be passed into the `_create_snapshot` function so looking into how to address this there as well